tempesta_fw/main.c

```diff
@@ -269,8 +271,7 @@ tfw_ctlfn_state_change(const char *old_state, const char *new_state)
 			TFW_LOG("Live reconfiguration of Tempesta.\n");
 		}
 
-		if (!(r = tfw_start(&tfw_mods)))
-			WRITE_ONCE(tfw_started, true);
+		r = tfw_start(&tfw_mods);
```

Comment:
Current `tfw_ctlfn_state_change()` is good that all the modification of `tfw_started` are there and the patch distributes the variable updates over the code. Why not just to change these 2 line like?
```C
r = tfw_start(&tfw_mods);
WRITE_ONCE(tfw_started, !r);
```

Reply:
In case of `true` this variant is ok. but in case of `false` - it disables `tfw_started` flag in both ( reconfiguration failed and start failed) events, and we need to disable it only in event of failed start. In failed reconfiguration case Tempesta remain in started state, so we need to preserve `tfw_started` flag in `true` value.